File: DoItTest/ClientApp/src/domain/tests/studentTestFilter.ts
import { DateTimePeriod } from "tools/dateTimePeriod";

export interface IStudentTestFilter {

    page: number,
    pageSize: number,
    testId: string | null,
    studentFIO: string | null,
    group: string | null,
    dateTimePeriod: DateTimePeriod
}

export function makeStudentTestFilter(filter: Partial<IStudentTestFilter>): IStudentTestFilter {
    const testId = null
    const studentFIO = null
    const group = null
    const dateTimePeriod = new DateTimePeriod(null, null)

    return { testId, studentFIO, group, dateTimePeriod, page: 1, pageSize: 10, ...filter };
}

export default IStudentTestFilter

File: DoItTest.Domain/Tests/StudentTestFilter.cs
namespace DoItTest.Domain.Tests
{
    public class StudentTestFilter
    {
        public Int32 Page { get; set; }
        public Int32 PageSize { get; set; }
        public String? Group { get; set; }
        public String? studentFIO { get; set; }
        public Guid? TestId { get; set; }
        public Guid? UserId { get; set; }
    }
}
